packages/payload/src/collections/endpoints/getFileFromURL.ts

```diff
@@ -0,0 +1,73 @@
+import type { PayloadHandler } from '../../config/types.js'
+
+import executeAccess from '../../auth/executeAccess.js'
+import { APIError } from '../../errors/APIError.js'
+import { Forbidden } from '../../errors/Forbidden.js'
+import { getRequestCollectionWithID } from '../../utilities/getRequestEntity.js'
+
+// If doc id is provided, it means we are updating the doc
+// /:collectionSlug/paste-url/:doc-id?src=:fileUrl
+
+// If doc id is not provided, it means we are creating a new doc
+// /:collectionSlug/paste-url?src=:fileUrl
+
+export const getFileFromURLHandler: PayloadHandler = async (req) => {
+  const { id, collection } = getRequestCollectionWithID(req, { optionalID: true })
+
+  if (!req.user) {
+    throw new Forbidden(req.t)
+  }
+
+  const config = collection?.config
+
+  if (id) {
+    // updating doc
+    const accessResult = await executeAccess({ req }, config.access.update)
+    if (!accessResult) {
+      throw new Forbidden(req.t)
+    }
+  } else {
+    // creating doc
+    const accessResult = await executeAccess({ req }, config.access?.create)
+    if (!accessResult) {
+      throw new Forbidden(req.t)
+    }
+  }
+  try {
+    if (!req.url) {
+      throw new APIError('Request URL is missing.', 400)
+    }
+
+    const { searchParams } = new URL(req.url)
+    const src = searchParams.get('src')
+
+    if (!src || typeof src !== 'string') {
+      throw new APIError('A valid URL string is required.', 400)
+    }
+
+    const validatedUrl = new URL(src)
+
+    // Fetch the file with no compression
+    const response = await fetch(validatedUrl.href, {
```

Comment:
The isURLAllowed function needs to be run before the fetch on the server side. Without this the security aspect of this is completely missed.